### PR TITLE
More realistic gasPrice calculation

### PR DESCRIPTION
**Pull request by adibas03**

### CODE REVIEW

File: src/Util.ts [hunk 1]
```diff
@@ -2,9 +2,45 @@ import BigNumber from 'bignumber.js';
...
+      }
+      return response.json()
+    })
+    .then((data: any) => { /* <-- data inferred as { data: T }*/
+      return data.data
+    })
+}
+
+const SERVICES: IServices = {//should return GWEI value of the gasPrice
+  '1': [
```
Comment (kosecki123) [inline src/Util.ts]: "1" is for network id I guess. Maybe introduce an enum to represent those networks?

File: src/Util.ts [hunk 2]
```diff
@@ -98,9 +99,58 @@ export default class W3Util {
...
     return new Promise((resolve, reject) => {
-      this.web3.eth.getGasPrice((e: any, r: any) => (e ? reject(e) : resolve(r)));
+      this.web3.version.getNetwork((e: any, r: any) => (e ? reject(e) : resolve(r)));
+    });
+  }
+
+  public fetchExternalApiNetworkPrice (networkId: string): Promise<any> {
+    const services = SERVICES[networkId];
+    return new Promise(async(resolve, reject) => {
+      for ( let s = 0; s < services.length; s++) {
```
Comment (lsaether) [inline src/Util.ts]: Could change these two lines to `for (let service of SERVICES[networkId]) { ... }`

File: src/Util.ts [hunk 3]
```diff
@@ -98,9 +99,58 @@ export default class W3Util {
...
+          if (s === (services.length-1)) {
+            reject(new Error('No result from services'));
+          }
+          continue;
+        }
+      }
+    });
+  }
+
+  public externalApiNetworkPrice (): Promise<BigNumber> {
```
Comment (lsaether) [inline src/Util.ts]: Could make this an `async` function which would be the same thing as returning a `Promise`.

File: src/Util.ts (hunk 2 above)
Comment (adibas03) [reply src/Util.ts]: There is a preference of `for` over `for ... of`, in terms of efficiency, and we have the es-lint rule for that as well

File: src/Util.ts (hunk 3 above)
Comment (adibas03) [reply src/Util.ts]: I was just following the patterns and rules we have implemented, actually.

**Pull request merged**